### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by morkot**
Originally I was thinking about dev container that would help to avoid pollution of dev workstation. Later I could make more product-ish container (smaller image without unneeded dev tools etc.)

### CODE REVIEW

**Comment by andrewmchen** [comment]
Hi @morkot, thanks for making a PR! What's the purpose of this Docker image? Do you use it to run the tests?

**Comment by morkot** [comment]
@andrewmchen tests are run only on the building stage. Then I use the container to run CLI in different environments without any collisions of python and libraries versions. Also, I plan to use it in Jenkins.

File: README.rst [hunk 1]
```diff
@@ -42,6 +42,18 @@ and follow the prompts. The second way is to use a access token generated inside
 Databricks. To configure the CLI to use an access token run ``dbfs configure --token``
 and follow the prompts.
 
+Using Docker
+------------
+
+To build image run
+``docker build -t $(pwd | sed 's@^.*/\([^$]*\)$@\1@') .``
```
Comment (andrewmchen) [inline README.rst]: could we use `basename $(pwd)` instead?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /usr/src/databricks-cli
+
+COPY . .
+
+RUN pip install --upgrade pip && \
+    pip install -r dev-requirements.txt
+
+RUN ./lint.sh && \
+    pip install . && \
+    pytest tests
```
Comment (andrewmchen) [inline Dockerfile]: Can we combine the two RUNs and add a CMD for `databricks`?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM python:2.7
+
+WORKDIR /usr/src/databricks-cli
+
+COPY . .
```
Comment (andrewmchen) [inline Dockerfile]: nit: would it be more conventional to specify the target path absolutely?

Comment (morkot) [reply Dockerfile]: Considering WORKDIR instruction above, I don't think so.

**Comment by morkot** [comment]
@andrewmchen are going to merge it? :)

**Comment by andrewmchen** [comment]
Done. Thanks for the contribution!

**Comment by morkot** [comment]
np. 

Do you know anything about API/CLI for DBFS mount of S3 buckets? 
https://github.com/databricks/databricks-cli/issues/10

**Pull request merged**